src/vanilla/injector.ts

```diff
@@ -703,7 +701,19 @@ function runMolecule(
   onMountImpl.push((fn: MountedCallback) => mountedCallbacks.add(fn));
   useImpl.push(use);
   let running = true;
-  trackingScopeGetter(InternalOnlyGlobalScope);
+
+  /**
+   * Create or reuse a unique internal global scope for this molecule.
+   * This ensures cleanup mechanisms work properly via the normal scope lifecycle.
+   */
+  if (!m[GlobalScopeSymbol]) {
+    const id = ++globalScopeId;
+    m[GlobalScopeSymbol] = createScope(Symbol(`bunshi.global.scope.${id}`), {
+      debugLabel: `Global Scope ${id}`,
+    });
+  }
+  trackingScopeGetter(m[GlobalScopeSymbol]);
```

Comment:
Fix type errors to fix the build.

```suggestion
  trackingScopeGetter(m[GlobalScopeSymbol]!);
```